pkg/odo/cli/component/common_push.go

```diff
@@ -330,3 +332,75 @@ func filterIgnores(filesChanged, filesDeleted, absIgnoreRules []string) (filesCh
 	}
 	return filesChangedFiltered, filesDeletedFiltered
 }
+
+// retrieveKubernetesDefaultNamespace tries to retrieve the current active namespace
+// to set as a default namespace
+func retrieveKubernetesDefaultNamespace() (string, error) {
+	// Get current active namespace
+	client, err := kclient.New()
+	if err != nil {
+		return "", err
+	}
+	return client.Namespace, nil
+}
```

Comment:
I feel this should be moved to the kClient package.

Reply:
Since it's a private function that we only use here, I think we can keep this here?

Reply:
But I think we can export this function like in the occlient package. It might be useful at many other places later on.